### ISSUE TITLE
[a11y]: ComboBox additional instructions needed for screen readers on how to clear the input.

### ISSUE THREAD

**Issue by tonydiaz** [carbon-design-system/carbon#21349]
### Package

@carbon/react

### Browser

Chrome

### Operating System

MacOS

### Package version

1.95.0

### React version

17.0.2

### Automated testing tool and ruleset

JAWS_Chrome

### Assistive technology

JAWS_Chrome

### Description

Component: Combobox clear input

Issue description: No instruction is provided for the screen reader users to clear the input with backspace or escape button. Keyboard users can't access the "X" to clear the input. They need additional instructions on how to do this. The titleText prop overrides the aria-label so you can't add additional information without it being displayed as a label as well.

Recommended fix: Provide instruction to the screen reader users that they can use backspace to erase the input. Allow use of the `aria-label` prop when using the `titleText` prop

User impact: Screen reader users cannot be able to clear the input they have provided.

WCAG Mapping: 3.2.2 labels and instructions

Level of Compliance: A

Severity: Medium

### WCAG 2.1 Violation

3.2.2 labels and instructions

### Reproduction/example

https://react.carbondesignsystem.com/?path=/docs/components-combobox--overview#overview

### Steps to reproduce

With the screen reader on:
- Using the keyboard add an option from a ComboBox. 
- Using the keyboard you are unable to access the "X" button to clear the input
- Need ability to add additional instructions to screen readers on how to clear the input. 

### Suggested Severity

None

### Code of Conduct

- [x] I agree to follow this project's [Code of Conduct](https://github.com/carbon-design-system/carbon/blob/f555616971a03fd454c0f4daea184adf41fff05b/.github/CODE_OF_CONDUCT.md)
- [x] I checked the [current issues](https://github.com/carbon-design-system/carbon/issues) for duplicate problems

**Comment by devadula-nandan** [comment]
the expected solution should be 
1) weather we need to announce the escape key behavior for clearing when we get the focus on combo-box after selection.
2) should the close button be focusable with right aria label "clear selection" in react?

need to be discussed with a11y team

**Comment by maradwan26** [comment]
We discussed this with the accessibility team and concluded the following:

The clear selection 'X' should not be tabbable. This is because it serves as a mouse affordance for an action that keyboard users can already do. A keyboard user can clear the input by clicking Esc or Ctrl + A then Delete/Backspace, a mouse user doesn’t have that same ability. The ‘X’ is there so that a mouse user can click on it, a keyboard user has other options to do that.

Making the ‘X’ tabbable would introduce an extra focus stop and make the flow less efficient: users would tab out of the input field to the clear button —> activate it —> tab back, compared to just clearing the value directly from the input.


Backspace, Esc, and Ctrl+A & deleting are considered standard behaviours for clearing input. Even if the user doesn't know they can hit Esc to clear, they know they can use Ctrl+A and Delete/Backspace. So announcing the option to use Esc to clear the input wasn’t considered necessary, since it is a standard behaviour.


The PR attached to this issue will align the web-components versions of this component to match this behaviour in React.
